From a35c2b3ac807e9cdea95873ababc35b7567ba5ad Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 13 Dec 2012 13:36:22 +0800
Subject: [PATCH 1/6] ENGR00224720 Add fsl kernel file header into libc

Add kernel header mxc_v4l2.h/mxcfb.h/ipu.h file link to kernel_imx.
Link the real linux/version.h to bionic kernel common header.

Change-Id: I8952e68c9fb09441174fbff64d8bf56b3b5b5946
Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 libc/kernel/common/linux/ipu.h      |  1 +
 libc/kernel/common/linux/mxc_v4l2.h |  1 +
 libc/kernel/common/linux/mxcfb.h    |  1 +
 libc/kernel/common/linux/version.h  | 21 +--------------------
 4 files changed, 4 insertions(+), 20 deletions(-)

diff --git a/libc/kernel/common/linux/ipu.h b/libc/kernel/common/linux/ipu.h
new file mode 120000
index 0000000..30662c1
--- /dev/null
+++ b/libc/kernel/common/linux/ipu.h
@@ -0,0 +1 @@
+../../../../../kernel_imx/include/linux/ipu.h
\ No newline at end of file
diff --git a/libc/kernel/common/linux/mxc_v4l2.h b/libc/kernel/common/linux/mxc_v4l2.h
new file mode 120000
index 0000000..1bff6cd
--- /dev/null
+++ b/libc/kernel/common/linux/mxc_v4l2.h
@@ -0,0 +1 @@
+../../../../../kernel_imx/include/linux/mxc_v4l2.h
\ No newline at end of file
diff --git a/libc/kernel/common/linux/mxcfb.h b/libc/kernel/common/linux/mxcfb.h
new file mode 120000
index 0000000..87102eb
--- /dev/null
+++ b/libc/kernel/common/linux/mxcfb.h
@@ -0,0 +1 @@
+../../../../../kernel_imx/include/linux/mxcfb.h
\ No newline at end of file
diff --git a/libc/kernel/common/linux/version.h b/libc/kernel/common/linux/version.h
deleted file mode 100644
index 469cb05..0000000
--- a/libc/kernel/common/linux/version.h
+++ /dev/null
@@ -1,20 +0,0 @@
-/****************************************************************************
- ****************************************************************************
- ***
- ***   This header was automatically generated from a Linux kernel header
- ***   of the same name, to make information necessary for userspace to
- ***   call into the kernel available to libc.  It contains only constants,
- ***   structures, and macros generated from the original header, and thus,
- ***   contains no copyrightable information.
- ***
- ***   To edit the content of this header, modify the corresponding
- ***   source file (e.g. under external/kernel-headers/original/) then
- ***   run bionic/libc/kernel/tools/update_all.py
- ***
- ***   Any manual change here will be lost the next time this script will
- ***   be run. You've been warned!
- ***
- ****************************************************************************
- ****************************************************************************/
-#define LINUX_VERSION_CODE 132626
-#define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
diff --git a/libc/kernel/common/linux/version.h b/libc/kernel/common/linux/version.h
new file mode 120000
index 0000000..344a6a2
--- /dev/null
+++ b/libc/kernel/common/linux/version.h
@@ -0,0 +1 @@
+../../../../../kernel_imx/include/linux/version.h
\ No newline at end of file
-- 
1.8.0

